widget/button.go

```diff
@@ -99,6 +99,7 @@ func (b *Button) CreateRenderer() fyne.WidgetRenderer {
 	b.background = canvas.NewRectangle(theme.ButtonColor())
 	b.background.CornerRadius = theme.InputRadiusSize()
 	tapBG := canvas.NewRectangle(color.Transparent)
+	tapBG.FillColor = &color.NRGBA{R: 0, G: 0, B: 0, A: 0}
```

Comment:
This seems strange. The `canvas.NewRectangle()` constructor already sets `FillColor`. Is this line really needed?

Reply:
Yes. The tap animation does not return the color to transparent (a nil color). It explicitly sets it to this value. The other alternative would be to alter the animation logic to return the color to nil at the end, but I think this would be a Pandora's box. I certainly do not know enough about animation to even begin to assess the feasibility of this kind of change.

Reply:
What I am trying to say is that you can replace `color.Transparent` on the line above instead of setting the field directly after.

However, if this is just a workaround for the animation doing strange things it seems like we ought to fix the animation instead? We are using `color.Transparent` all over the place and if we need this workaround everywhere then something seems very wrong.

Thoughts? @andydotxyz

Reply:
Yes, I see. I didn't realize that the value supplied to the NewRectangle() function was the background color - I thought it was the line (edge) color.

With respect to the use of color.Transparent, I suspect it is only an issue with animation. I'll let you and @andydotxyz discuss and draw conclusions - I'll go with whatever you decide.

Reply:
Colour.Transparent is not `nil`... 

https://cs.opensource.google/go/go/+/refs/tags/go1.21.1:src/image/color/color.go%3Bl=345

not sure what nil is being referenced!

Reply:
Indeed. That was my though as well

Reply:
Sorry, I was wrong about transparent being nil: it is a struct with only the A value specified. In any case, I found a better fix. I modified the Button animation so that if the fade (i.e. the A value of the color) reaches zero, the assigned value for the bg.FillColor is color.Transparent, which was the default initial value. Updated test results accordingly.